zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala

```diff
@@ -469,7 +469,7 @@ class IncrementalCompilerImpl extends IncrementalCompiler {
         extra
       )
       def prevResult = CompileResult.of(prev, config.currentSetup, false)
-      if (skip) prevResult
+      if (skip && !incrementalOptions.pipelining) prevResult
```

Comment:
Why this change?

Reply:
This is so the notification would fire up.